### PR TITLE
Visitor can see the articles according to their categories

### PR DESCRIPTION
**Pull request by Odeane**
### User Story
```
As a visitor 
in order to get relevant news
I would like to see the articles in categories
```
## Changes proposed in this pull request:
* User can now view articles based on their category

## What I have learned working on this feature:
* We learned how to use react router

## Packages added
- axios
- react-router-dom
## Screenshots (Client)
Screenshot 2020-08-10 at 18 27 19

Screenshot 2020-08-10 at 18 27 45

### CODE REVIEW

File: src/Components/Pages/EconomyPage.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          if (article.category === 'economy') {
+            return (
+              <div className='article-list'>
+                <h1 id="">{article.title}</h1>
+                <h2 id="">{article.lead}</h2>
+                <button id="button">Read more</button>
+              </div>
+            )
+          } else {
+            return 'Unable to show categories'
```
Comment (viktorsommar) [inline src/Components/Pages/EconomyPage.jsx]: Shouldn't this say "Unable to show articles"? Or "No articles were found"?

File: src/Components/Pages/LifestylePage.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            if (article.category === 'lifestyle') {
+              return (
+                <div className='article-list'>
+                  <h1 id="">{article.title}</h1>
+                  <h2 id="">{article.lead}</h2>
+                  <button id="button">Read more</button>
+                </div>
+              )
+            } else {
+              return 'Unable to show categories'
```
Comment (viktorsommar) [inline src/Components/Pages/LifestylePage.jsx]: Shouldn't this say "Unable to show articles"? Or "No articles were found"?

File: src/Components/Pages/SportsPage.jsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            if (article.category === 'sports') {
+              return (
+                <div className='article-list'>
+                  <h1 id="">{article.title}</h1>
+                  <h2 id="">{article.lead}</h2>
+                  <button id="button">Read more</button>
+                </div>
+              )
+            } else {
+              return 'Unable to show categories'
```
Comment (viktorsommar) [inline src/Components/Pages/SportsPage.jsx]: Shouldn't this say "Unable to show articles"? Or "No articles were found"?

**Review by viktorsommar — APPROVED** [review]
Left some comments about the message about the categories conditional

**Review by Felix-Willborg — APPROVED** [review]

File: src/Components/Articles.jsx [hunk 4]
```diff
@@ -5,14 +5,35 @@ import ArticleContent from './ArticleContent'
...
+    if (this.props.history.location.pathname === "/") {
+      response = await axios.get(`/articles`);
+    } else {
+      response = await axios.get(`/articles`,
+        {
+          params: this.props.match.params
+        }
+      );
+    }
+    console.log(response)
```
Comment (oliverochman) [inline src/Components/Articles.jsx]: plz remove console log

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: src/Components/Articles.jsx (hunk 4 above)
Comment (Odeane) [reply src/Components/Articles.jsx]: resolved

File: src/Components/Articles.jsx [hunk 5]
```diff
@@ -5,14 +5,34 @@ import ArticleContent from './ArticleContent'
...
 
+  getArticles = async () => {
+    let response
+    if (this.props.history.location.pathname === "/") {
+      response = await axios.get(`/articles`);
+    } else {
+      response = await axios.get(`/articles`,
+        {
+          params: this.props.match.params
+        }
```
Comment (emtalen) [inline src/Components/Articles.jsx:30]: I would try this request while running the backend to make sure it is working, if you haven't done that. To make sure you don't get the params keyword twice in the request

**Pull request merged**